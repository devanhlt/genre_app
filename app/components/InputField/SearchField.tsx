import { appColors, iconSizes } from "app/theme"
import React from "react"
import { View } from "react-native"
import { PhosphorIcon } from "../Icon/PhosphorIcon"
import { TextField, TextFieldProps } from "./TextField"

const $searchPresets = {
  plat: {
    activeBorderColor: appColors.palette.transparent,
    inputWrapperStyle: {
      backgroundColor: appColors.components.tag.containerBlue,
      borderColor: appColors.palette.transparent,
    },
  } as TextFieldProps,
  outline: { activeBorderColor: appColors.palette.red300 } as TextFieldProps,
}

type Presets = keyof typeof $searchPresets

export interface SearchFieldProps extends Omit<TextFieldProps, "ref"> {
  /**
   * One of the different types of search field presets.
   */
  preset?: Presets
}

function SearchField(props: SearchFieldProps) {
  const [value, setValue] = React.useState(props.value || "")

  const onClear = () => {
    setValue("")
    props?.onChangeText("")
  }

  const onChangeText = (text: string) => {
    setValue(text)
    props?.onChangeText(text)
  }

  const $searchPresetsStyles = $searchPresets[props?.preset || "outline"]
  return (
    <TextField
      {...props}
      {...$searchPresetsStyles}
      LeftAccessory={(props) => (
        <View {...props}>
          <PhosphorIcon name="MagnifyingGlass" size={iconSizes.xsmall} />
        </View>
      )}
      value={value}
      onChangeText={onChangeText}
      onClear={onClear}
    />
  )
}

export default SearchField
