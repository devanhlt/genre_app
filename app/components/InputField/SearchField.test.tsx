import React from "react"

import renderer from "react-test-renderer"
import SearchField from "./SearchField"

jest.mock("react-native-gesture-handler", () => undefined)

describe("<SearchField />", () => {
  it("renders with plat preset correctly", () => {
    const tree = renderer.create(<SearchField preset="plat" />)
    expect(tree).toMatchSnapshot()
  })

  it("renders with outline preset correctly", () => {
    const tree = renderer.create(<SearchField preset="outline" />)
    expect(tree).toMatchSnapshot()
  })

  it("should changed text is passed as an argument to the callback handler", () => {
    const mockCallback = jest.fn(() => null)
    const tree = renderer.create(<SearchField preset="outline" onChangeText={mockCallback} />)
    expect(tree).toMatchSnapshot()
  })
})
