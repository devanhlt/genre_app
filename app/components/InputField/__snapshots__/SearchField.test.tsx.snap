// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> renders with outline preset correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#A1ABBB",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "overflow": "hidden",
          "padding": 8,
        },
        false,
        false,
        false,
        undefined,
        {
          "paddingStart": 0,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      editable={true}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginStart": 8,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        color="#000"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        size={16}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            [
              undefined,
              undefined,
              {},
            ],
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        tintColor="#000"
        vbHeight={256}
        vbWidth={256}
        width={16}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGRect
            fill={null}
            height="256"
            propList={
              [
                "fill",
              ]
            }
            width="256"
            x="0"
            y="0"
          />
          <RNSVGCircle
            cx="116"
            cy="116"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            r="84"
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="16"
          />
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="16"
            x1="175.4"
            x2="224"
            y1="175.4"
            y2="224"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <TextInput
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8693A8"
      preset="outline"
      style={
        [
          {
            "alignSelf": "stretch",
            "color": "#001B48",
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 0,
            "marginHorizontal": 4,
            "marginVertical": 4,
            "paddingHorizontal": 0,
            "paddingVertical": 0,
            "textAlign": "auto",
            "textAlignVertical": "center",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      textAlignVertical="top"
      underlineColorAndroid="#FBFBFB"
      value=""
    />
  </View>
</View>
`;

exports[`<SearchField /> renders with plat preset correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#A1ABBB",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "overflow": "hidden",
          "padding": 8,
        },
        false,
        false,
        false,
        undefined,
        {
          "paddingStart": 0,
        },
        undefined,
        {
          "backgroundColor": "#F2F5FB",
          "borderColor": "rgba(0, 0, 0, 0)",
        },
      ]
    }
  >
    <View
      editable={true}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginStart": 8,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        color="#000"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        size={16}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            [
              undefined,
              undefined,
              {},
            ],
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        tintColor="#000"
        vbHeight={256}
        vbWidth={256}
        width={16}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGRect
            fill={null}
            height="256"
            propList={
              [
                "fill",
              ]
            }
            width="256"
            x="0"
            y="0"
          />
          <RNSVGCircle
            cx="116"
            cy="116"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            r="84"
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="16"
          />
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="16"
            x1="175.4"
            x2="224"
            y1="175.4"
            y2="224"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <TextInput
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8693A8"
      preset="plat"
      style={
        [
          {
            "alignSelf": "stretch",
            "color": "#001B48",
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 0,
            "marginHorizontal": 4,
            "marginVertical": 4,
            "paddingHorizontal": 0,
            "paddingVertical": 0,
            "textAlign": "auto",
            "textAlignVertical": "center",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      textAlignVertical="top"
      underlineColorAndroid="#FBFBFB"
      value=""
    />
  </View>
</View>
`;

exports[`<SearchField /> should changed text is passed as an argument to the callback handler 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#A1ABBB",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "overflow": "hidden",
          "padding": 8,
        },
        false,
        false,
        false,
        undefined,
        {
          "paddingStart": 0,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      editable={true}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginStart": 8,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        color="#000"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        size={16}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            [
              undefined,
              undefined,
              {},
            ],
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        tintColor="#000"
        vbHeight={256}
        vbWidth={256}
        width={16}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGRect
            fill={null}
            height="256"
            propList={
              [
                "fill",
              ]
            }
            width="256"
            x="0"
            y="0"
          />
          <RNSVGCircle
            cx="116"
            cy="116"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            r="84"
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="16"
          />
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="16"
            x1="175.4"
            x2="224"
            y1="175.4"
            y2="224"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <TextInput
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8693A8"
      preset="outline"
      style={
        [
          {
            "alignSelf": "stretch",
            "color": "#001B48",
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 0,
            "marginHorizontal": 4,
            "marginVertical": 4,
            "paddingHorizontal": 0,
            "paddingVertical": 0,
            "textAlign": "auto",
            "textAlignVertical": "center",
          },
          false,
          false,
          undefined,
          undefined,
        ]
      }
      textAlignVertical="top"
      underlineColorAndroid="#FBFBFB"
      value=""
    />
  </View>
</View>
`;
